### ISSUE TITLE
Trying to get property of non-object in content.php on line 350 

### ISSUE THREAD

**Issue by rhurtik** [strangerstudios/paid-memberships-pro#488]
Please use the bellow fix : 

```
diff --git a/wp-content/plugins/paid-memberships-pro/includes/content.php b/wp-content/plugins/paid-memberships-pro/includes/content.php
index fc69b42..6e6156b 100644
--- a/wp-content/plugins/paid-memberships-pro/includes/content.php
+++ b/wp-content/plugins/paid-memberships-pro/includes/content.php
@@ -60,7 +60,7 @@ function pmpro_has_membership_access($post_id = NULL, $user_id = NULL, $return_m
                else
                {
                        //are any of the post categories associated with membership levels? also check the memberships_pages table
-                       $sqlQuery = "(SELECT m.id, m.name FROM $wpdb->pmpro_memberships_categories mc LEFT JOIN $wpdb->pmpro_membership_levels m ON mc.membership_id = m.id WHERE mc.category_id IN(" . implode(",", $post_categories) . ") AND m.id IS NOT NULL) UNION (SELECT m.id, m.name FROM $wpdb->pmpro_memberships_pages mp LEFT JOIN $wpdb->pmpro_membership_levels m ON mp.membership_id = m.id WHERE mp.page_id = '" . $mypost->ID . "')";
+                       $sqlQuery = "(SELECT m.id, m.name FROM $wpdb->pmpro_memberships_categories mc LEFT JOIN $wpdb->pmpro_membership_levels m ON mc.membership_id = m.id WHERE mc.category_id IN(" . implode(",", $post_categories) . ") AND m.id IS NOT NULL) UNION (SELECT m.id, m.name FROM $wpdb->pmpro_memberships_pages mp LEFT JOIN $wpdb->pmpro_membership_levels m ON mp.membership_id = m.id WHERE mp.page_id = '" . $mypost->ID . "' AND m.id IS NOT NULL)";
                }
        }

```

**Comment by strangerstudios** [comment]
Thanks for the heads up. We believe all warnings should be fixed, so will get on this ASAP.

Another fix might be appropriate. Perhaps we can test the result from the query vs adjust the query. Out of context,  " AND m.id IS NOT NULL" should maybe be " AND mp.id IS NOT NULL" considering the second part of that UNION is querying on mp. I'll take a look and figure it out.

**Comment by strangerstudios** [comment]
I think this is the better fix for this warning and will be included in 1.9.4:
https://github.com/strangerstudios/paid-memberships-pro/commit/6419f9e6061a4f543611e03f88474e77e0a4e586

I'm going to leave this issue open though because there might be a different issue going on. It seems like the pmpro_has_membership_access check is seeing rows in the memberships_categories or memberships_pages tables for levels that don't exist anymore. When you delete a level, a function runs to clean up after itself, but that might be broken.

Another possible issue could be with certain multisite setups where the data is getting mixed up. Might be worth testing.

My patch might fix the warning, but you might still have bad data in your DB. If you know why/how that data got bad, let me know so we can look into it.

Thanks.

**Comment by ideadude** [comment]
Closing this issue in favor of this one: https://github.com/strangerstudios/paid-memberships-pro/issues/1552
